messagingpush/src/main/java/io/customer/messagingpush/activity/NotificationClickReceiverActivity.kt

```diff
@@ -43,98 +34,23 @@ class NotificationClickReceiverActivity : Activity(), TrackableScreen {
     }
 
     private fun handleIntent(data: Intent?) {
-        kotlin.runCatching {
-            val payload: CustomerIOParsedPushPayload? =
-                data?.extras?.parcelable(NOTIFICATION_PAYLOAD_EXTRA)
-            if (payload == null) {
-                logger.error("Payload is null, cannot handle notification intent")
+        if (data == null) {
+            // This should never happen ideally
+            logger.error("Intent is null, cannot process notification click")
+        } else {
+            val sdkInstance = CustomerIO.instanceOrNull(context = this)
+            if (sdkInstance == null) {
+                logger.error("SDK is not initialized, cannot handle notification intent")
             } else {
-                processNotificationIntent(payload = payload)
+                sdkInstance.diGraph.pushMessageProcessor.processNotificationClick(
```

Comment:
We can. But I avoided because I don't think this adds a lot of value. It mostly relies on Android OS lifecycle and couple of other methods that are already being tested. Also, it requires setting up Activity lifecycle for testing which I plan to improve later to gain more confidence in testing. However, if you think we should still cover this, I can write a test for it.

Reply:
I think it still alots value because the whole logic revolves around what flags to expect and what behaviour it would display.

Reply:
I have pushed more tests, hope it helps.

PS: Github is having issues in PRs, so it might take some time to show new changes.